src/services/TestResultsService.ts

```diff
@@ -309,14 +307,21 @@ export class TestResultsService {
                       const anvDateForCompare = regOrFirstUseDate ? dateFns.endOfDay(this.lastDayOfMonthInNextYear(regOrFirstUseDate)).toISOString() : undefined;
                       // If anniversaryDate is not populated in tech-records OR test date is 2 months or more before the Registration/First Use Anniversary for HGV/TRL
                       console.log(`Current date: ${new Date()}, annv Date: ${anvDateForCompare}`);
-                      if (!anvDateForCompare || dateFns.isBefore(new  Date(), dateFns.subMonths(anvDateForCompare, 2))) {
+                      if (!anvDateForCompare || dateFns.isBefore(new Date(), dateFns.subMonths(anvDateForCompare, 2))) {
                         testType.testExpiryDate = this.lastDayOfMonthInNextYear(new Date()).toISOString();
                         console.log(`Setting expiryDate: ${testType.testExpiryDate}`);
                       } else {
                         // less than 2 months then set expiryDate 1 year after the Registration/First Use Anniversary date
                         testType.testExpiryDate = dateFns.addYears(anvDateForCompare, 1).toISOString();
                         console.log(`Setting expiryDate as 1yr from RegDate: ${testType.testExpiryDate}`);
                       }
+                    } else if (this.isAnnualTestRetestTestType(testType) && dateFns.isEqual(mostRecentExpiryDateOnAllTestTypesBySystemNumber, new Date(1970, 1, 1))) {
+                        const registrationFirstUseAnniversaryDate = dateFns.addYears(dateFns.lastDayOfMonth(regOrFirstUseDate!), 1);
+                        if (!regOrFirstUseDate || dateFns.isBefore(dateFns.addMonths(new Date(), 2), registrationFirstUseAnniversaryDate) || dateFns.isEqual(dateFns.addMonths(new Date(), 2), registrationFirstUseAnniversaryDate)) {
```

Comment:
I don't understand the third condition on this. It looks the same as the second - I would just compare to "the end of the registration/firstUse anniversary date day" - there's an dateFns.endOfDay function for that.

Reply:
it's for further enforcing the case: "test date exactly 2 months before the registration anniversary/ first use anniversary". Apparently the 2nd condition is enough, but given the latest issues we had with dateFns I considered the 3rd wouldn't hurt.

Reply:
fixed with your suggestion